kbase-extension/static/kbase/js/common/jobs.js

```diff
@@ -153,12 +166,17 @@ define(['common/errorDisplay', 'common/format', 'common/html', 'common/ui', 'uti
      * @returns {boolean} true|false
      */
     function isValidBackendJobStateObject(backendJobState) {
-        return !!(
+        const required = ['jobState'];
+        if (
             backendJobState &&
             Utils.objectToString(backendJobState) === 'Object' &&
+            required.every((prop) => prop in backendJobState) &&
             backendJobState.jobState &&
             isValidJobStateObject(backendJobState.jobState)
-        );
+        ) {
+            return true;
+        }
+        return isValidJobErrorObject(backendJobState);
```

Comment:
For all these checks, the message content (which is what these functions act on) can also be an error object.